airbyte-cdk/bulk/toolkits/load-low-code/src/main/kotlin/io/airbyte/cdk/load/writer/DeclarativeLoaderState.kt

```diff
@@ -0,0 +1,53 @@
+package io.airbyte.cdk.load.writer
+
+import io.airbyte.cdk.load.http.HttpRequester
+import io.airbyte.cdk.load.http.InterpolableResponse
+import io.airbyte.cdk.load.http.Response
+import io.airbyte.cdk.load.http.decoder.JsonDecoder
+import io.airbyte.cdk.load.http.getBodyOrEmpty
+import io.airbyte.cdk.load.message.DestinationRecordRaw
+import io.airbyte.cdk.load.writer.batch.ResponseBodyBuilder
+import io.airbyte.cdk.load.writer.rejected.RejectedRecordsBuilder
+import io.github.oshai.kotlinlogging.KotlinLogging
+
+private val logger = KotlinLogging.logger {}
+
+
+class DeclarativeLoaderState(
+    private val httpRequester: HttpRequester,
+    private val responseBodyBuilder: ResponseBodyBuilder,
+    private val rejectedRecordsBuilder: RejectedRecordsBuilder,
+    private val responseDecoder: JsonDecoder = JsonDecoder()
+) : AutoCloseable {
+
+    // TODO notify RejectedRecordStrategy of a new record
+    fun accumulate(record: DestinationRecordRaw) {
+        responseBodyBuilder.accumulate(record)
+        rejectedRecordsBuilder.accumulate(record)
+    }
+
+    fun isFull(): Boolean = responseBodyBuilder.isFull()
```

Comment:
The flow where `isFull` is called on each record is a bit flawed as what will always happen is that we accumulate until `isFull` so `isFull` will always need to account for a buffer. For `RequestMemoryBatchSizeStrategy`, this is even more annoying because we could have 10 kb on a limit of 12 kb and we receive a record for 5 kb. This case illustrate that the buffer might need to be very big. As some point, maybe the design will need to change to `wouldBeFull(record)` to account for that